### PR TITLE
New Auth Flow with Firebase Functions Submodule

### PR DESCRIPTION
**Pull request by pauljohanneskraft**
# New Auth Flow with Firebase Functions Submodule

## :recycle: Current situation & Problem
Combines #86 with #89 to hopefully resolve issues faster.

## :gear: Release Notes 
*Add a bullet point list summary of the feature and possible migration guides if this is a breaking change so this section can be added to the release notes.*
*Include code snippets that provide examples of the feature implemented or links to the documentation if it appends or changes the public interface.*

## :books: Documentation
*Please ensure that you properly document any additions in conformance to [Spezi Documentation Guide](https://github.com/StanfordSpezi/.github/blob/main/DOCUMENTATIONGUIDE.md).*
*You can use this section to describe your solution, but we encourage contributors to document your reasoning and changes using in-line documentation.* 

## :white_check_mark: Testing
*Please ensure that the PR meets the testing requirements set by CodeCov and that new functionality is appropriately tested.*
*This section describes important information about the tests and why some elements might not be testable.*

### Code of Conduct & Contributing Guidelines 

By submitting creating this pull request, you agree to follow our [Code of Conduct](https://github.com/StanfordBDHG/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordBDHG/.github/blob/main/CONTRIBUTING.md):
- [x] I agree to follow the [Code of Conduct](https://github.com/StanfordBDHG/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordBDHG/.github/blob/main/CONTRIBUTING.md).

### CODE REVIEW

**Comment by pauljohanneskraft** [comment]
This PR is currently blocked by the following PRs:

https://github.com/StanfordSpezi/SpeziAccount/pull/75
https://github.com/StanfordSpezi/SpeziFirebase/pull/46

After they have been merged, we can release respective versions and then specify them in ENGAGE-HF-iOS.

File: ENGAGEHFUITests/HeartHealth/HeartHealthUITests.swift
Comment (pauljohanneskraft) [inline ENGAGEHFUITests/HeartHealth/HeartHealthUITests.swift:244]: This is worth looking into, maybe

Comment (pauljohanneskraft) [reply ENGAGEHFUITests/HeartHealth/HeartHealthUITests.swift:244]: https://github.com/StanfordSpezi/SpeziDevices/issues/13

**Review by PSchmiedmayer — APPROVED** [review]
Amazing for all the work here to incorporate @nriedman's work and all the account improvements; looks great!

**Pull request merged**
